### DISCUSSION TITLE
it seems Select Distinct generated query doesn't use LIMIT

### DISCUSSION THREAD

**Discussion by meotimdihia** [prisma/prisma#6227] [Q&A]
it seems Select Distinct generate Query doesn't use LIMIT 

https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing#select-distinct

I'm using the newest Prisma version.

**Comment by ryands17** [comment]
@meotimdihia 👋 

Are you asking about the `limit` keyword? If so, then the query uses it:

![image](https://user-images.githubusercontent.com/19697099/112301412-ea270500-8cbf-11eb-9979-bc65a935e963.png)

Also as mentioned in the doc, it doesn't use the `distinct` keyword but selects distinct values after fetching the records.

**Reply by meotimdihia** [reply]
it is strange. it doesn't work on my side: 

```javascript
await prisma.md_chapters.findMany({
  distinct: ['md_comic_id'],
  select: {
    id: true,
    md_comic_id: true,
  },
  take: 100
});
```

```
prisma:info Fetched a connection from the pool
{
  timestamp: 2021-03-24T11:27:43.122Z,
  query: 'SELECT "public"."md_chapters"."id", "public"."md_chapters"."md_comic_id" FROM "public"."md_chapters" WHERE 1=1 ORDER BY "public"."md_chapters"."id" ASC OFFSET $1',
  params: '[0]',
  duration: 3812,
  target: 'quaint::connector::metrics'
}
```
![wezterm-gui_2021-03-24_18-29-07](https://user-images.githubusercontent.com/300961/112303427-d8992980-8cce-11eb-8e99-480351fb6653.png)

**Reply by ryands17** [reply]
What version of Prisma are you using? If you're not on the latest could you update and check again?

**Reply by meotimdihia** [reply]
Hi, I checked 
"@prisma/client": "^2.19.0",
"@prisma/cli": "^2.19.0",
"@prisma": "^2.19.0",

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["orderByRelation"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

**Reply by ryands17** [reply]
@meotimdihia Could you try your model and query in this sample repo and check? In this one I am getting the correct values with `limit` applied.

[test-prisma.zip](https://github.com/prisma/prisma/files/6202924/test-prisma.zip)

**Reply by meotimdihia** [reply]
I tried to copy your files in my project. And it works perfectly.
I think Progresql could be the problem.

**Reply by ryands17** [reply]
Yup this seems to be with Postgres although the data is returned properly. I would suggest creating an [issue here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) so that we can look into this but you will get the correct data :)

![image](https://user-images.githubusercontent.com/19697099/112483011-a35c0c80-8d9e-11eb-8269-f8042601fcf9.png)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
